src/manager/node.c

```diff
@@ -433,6 +434,46 @@ static int node_on_match_proxy_new(sd_bus_message *m, void *userdata, UNUSED sd_
         return proxy_monitor_start_unit(monitor, unit_name);
 }
 
+static int node_on_match_proxy_removed(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
+        Node *requesting_node = userdata;
+
+        const char *node_name = NULL;
+        const char *unit_name = NULL;
+
+        int r = sd_bus_message_read(m, "ss", &node_name, &unit_name);
+        if (r < 0) {
+                hirte_log_errorf("Invalid arguments in ProxyRemoved signal: %s", strerror(-r));
+                return r;
+        }
+        hirte_log_debugf(
+                        "Matched ProxyRemoved signal from node '%s' for unit '%s' on node '%s'",
+                        requesting_node->name,
+                        unit_name,
+                        node_name);
+
+        Manager *manager = requesting_node->manager;
+
+        Node *providing_node = manager_find_node(manager, node_name);
+        if (providing_node == NULL) {
+                hirte_log_errorf("Could not find node with name '%s'", node_name);
+                return -ENXIO;
+        }
+        if (!node_is_online(providing_node)) {
+                hirte_log_errorf("Node '%s' is offline", node_name);
+                return -ENXIO;
+        }
+
+        r = sd_bus_message_append(m, "s", "replace");
+        if (r < 0) {
+                hirte_log_errorf("Failed to add mode to ProxyRemoved stop unit message: %s", strerror(-r));
+                return r;
+        }
+        (void) sd_bus_message_rewind(m, true);
+        (void) sd_bus_message_skip(m, "s"); /* skip node name */
+
+        return node_run_unit_lifecycle_method(m, providing_node, "stop", "StopUnit");
```

Comment:
This is not at all right. If some proxy on some node no longer depends on a remove service that doesn't mean we should stop it. Some other proxy, or a local service on the node may very well still depend on it.  For now, all we need to do is clean up the proxy monitor in the manager because we no longer need to track the remote service being stopped.

There has been discussions of how we can handle remote stop-dependencies, but this is a very tricky area, rife with race conditions and semantical details. Lets just ignore it for now.